src/main/java/io/stargate/sgv2/jsonapi/api/request/HeaderBasedEmbeddingCredentialResolver.java

```diff
@@ -0,0 +1,36 @@
+package io.stargate.sgv2.jsonapi.api.request;
+
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Implementation to resolve the embedding api key, access id and secret id from the request header.
+ */
+public class HeaderBasedEmbeddingCredentialResolver implements EmbeddingCredentialResolver {
+  private final String tokenHeaderName;
+  private final String accessIdHeaderName;
+  private final String secretIdHeaderName;
+
+  public HeaderBasedEmbeddingCredentialResolver(
+      String tokenHeaderName, String accessIdHeaderName, String secretIdHeaderName) {
+    Objects.requireNonNull(tokenHeaderName, "Token header name cannot be null");
```

Comment:
Can combine assignment with tests so:

```
this.tokenHeaderName = Objects.requireNonNull(tokenHeaderName, "Token header name cannot be null");
```

instead of separate statements.